avm/ptn/ai-platform/baseline/main.bicep

```diff
@@ -117,55 +337,71 @@ module keyVault 'br/public:avm/res/key-vault/vault:0.6.2' = {
 module storageAccount 'br/public:avm/res/storage/storage-account:0.11.0' = {
   name: '${uniqueString(deployment().name, location)}-storage'
   params: {
-    name: storageAccountSettings.?name ?? 'st${name}'
+    name: storageAccountConfiguration.?name ?? 'st${name}'
     location: location
-    skuName: storageAccountSettings.?sku ?? 'Standard_RAGZRS'
+    skuName: storageAccountConfiguration.?sku ?? 'Standard_RAGZRS'
     enableTelemetry: enableTelemetry
     allowBlobPublicAccess: false
-    allowSharedKeyAccess: storageAccountSettings.?allowSharedKeyAccess ?? false
-    defaultToOAuthAuthentication: !(storageAccountSettings.?allowSharedKeyAccess ?? false)
+    allowSharedKeyAccess: storageAccountConfiguration.?allowSharedKeyAccess ?? false
+    defaultToOAuthAuthentication: !(storageAccountConfiguration.?allowSharedKeyAccess ?? false)
     publicNetworkAccess: 'Disabled'
     networkAcls: {
       defaultAction: 'Deny'
       bypass: 'AzureServices'
     }
+    privateEndpoints: subnetResourceId != null
+      ? map(items(storagePrivateDnsZones), zone => {
+          name: 'pep-${zone.value}-${name}'
+          customNetworkInterfaceName: 'nic-${zone.value}-${name}'
+          service: zone.value
+          subnetResourceId: subnetResourceId
+          privateDnsZoneResourceIds: [resourceId('Microsoft.Network/privateDnsZones', zone.key)]
+        })
+      : null
     roleAssignments: [
       {
         principalId: managedIdentity.properties.principalId
         roleDefinitionIdOrName: 'Contributor'
+        principalType: 'ServicePrincipal'
```

Comment:
Don't want to stop this going through - but more of a question - why does it need Contributor - I would have thought the newer fine grained Storage x Contributor should do what we need

Reply:
I'm not sure, but it's recommended in the docs, and I'd be quite hesitant to change this. I imagine we would run into other issues.